### DISCUSSION TITLE
Image loading is super slow with next/image

### DISCUSSION THREAD

**Discussion by Aviisek** [vercel/next.js#21294] [Help]
I am using next/image as described 

```
<Image
                alt='alt '
                src={getImage()}
                layout='fill'
                objectFit='cover'
                objectPosition='right bottom'
            />
```
but its superslow while loading the page 
![image](https://user-images.githubusercontent.com/34102908/104946240-00f86500-59ba-11eb-9836-d43175623500.png)

what could be the issue or its expected behaviour because of optimisation on demand ?

**Comment by Sandstedt** [comment]
@Aviisek Sounds like a server problem. What server and version are you using? How long waiting time do you have on other images? If you look at the response header, does it has a cache policy set? Working on converting our framework to using next/image, and has been very fast this far at least.

**Reply by Aviisek** [reply]
@Sandstedt I am running it on node 10 and images are placed in public folder . when I use simple <img> tag it loads quickly. But for some reason its stuck in pending state, as I noticed it in network tab, before loading the image when I am using the next/image . you can see the TTFB is quite late.

**Reply by Sandstedt** [reply]
Well I'm not a back-end developer or contributor to next.js, so cant help you more I'm afraid. But would look into if there are any other server processes that's running and taking up resources, or if Node is spiking the CPU or something. "Works on my computer" :)

**Reply by a12791602** [reply]
i think the next/image is not good design to be honest , it cause a lot odd problems , such like take up lots of CPU usage ` loading slowly

**Reply by cplpearce** [reply]
I'm having the same problems, reverting back to `<img />` for the meantime. 

For anyone viewing this, I had a slight improvement with `priority={true}` and `loading="eager"`, and quality from the default cache of 60 (`quality={60}`) but even then some images took about 3 seconds to load.

**Reply by caillou** [reply]
I have the feeling, that the images are not kept in the service cache for a long time. This leads to initial load times that are super slow. Once the image is in Vercel's cache, load times are much better.

Maybe this only happens on free-tiers ...

**Reply by Aviisek** [reply]
Just to update: It didn't work for me, It was loading images super slow.  So I decided to use CDN.

**Reply by Sandstedt** [reply]
Yeah we have switched to a ImgIx / S3 bucket solution. So have build a custom image loader to make that connection work (was quite easy, just maybe 5 lines of code). The build in loader for ImgIx didn't give us the ability to set crop's etc.

We saw the exact same problem as you @Aviisek while upgradeing to a  never next.js version. 10.2 partially solved it, but was still too slow and took too much resources. It think it was around 10.0.6 they change next/image image rendering to another package to support ARM processors, and that is when things went downhill. So users has been able to downgrade to an earlier version of next.js and got it to work fine.

But having a CDN solutions for images seems like a good idea anyway. Or just pay for Vercels hosting that has CDN and fast servers build in (clever there).

**Reply by Sandstedt** [reply]
Just to clarify, the build in solution works great as long as you go through and generate all images for each sizes needed by hand, and make sure the cache-header is set to a year or so (else they will be deleted fast), and make sure your CI system doesn't delete the .next/image folder and clearing all previous generated images when deploying a new version of your app/site.

**Reply by caillou** [reply]
@Sandstedt could you elaborate how I would go about this? How do I konw, which sizes are needed? Would I commit the chached images into the `.next/image` folder? How can I set the cache headers, when using Vercel to deploy my application?

**Reply by aaroncrawford** [reply]
Glad I'm not the only one seeing very bad performance when using next/image.  Sometimes the images pop in, sometimes they get stuck indefinitely pending.  Something is definitely wrong here.  I'm also on a free account at the moment while I test it out.  Is this happening to paid plans as well?

**Reply by Fenntasy** [reply]
> Is this happening to paid plans as well?

Yes

**Reply by aaroncrawford** [reply]
Thanks for the update @Fenntasy.  For anyone else experiencing this now, the setup with imgix was very easy.  Just setup a web folder in imgix that points to your site, then setup the loader per the Next docs and everything will work as usual with next/image.

I tried Cloudinary in the past but their pricing was outrageous.  When we switched to imgix our bill dropped by 80% with all of our clients.  Haven't noticed any slow/misses from Imgix in all the months we've been using them.

**Reply by Sandstedt** [reply]
@caillou Just follow the documentation for custom loaders and you should probably understand. But you don't control the different sizes in the loader, you only get a prop with the width of the one that the browsers demands sort of. So there is some layers between there that you cant control, and you can't send in custom props. But you can create multiple custom loaders, ex one for each aspect ratio if you want. And then attach this custom loader directly to the <Image> components.

You can also read the source code for a bit more clarification how it works if you're interested: https://github.com/vercel/next.js/blob/c76170e82929bcf960ae577280eb5cb81f9919bd/packages/next/client/image.tsx

**Reply by eddsaura** [reply]
Hey people, thanks to everyone. I discovered imigix here and now I am gonna try. First time doing cdn stuff tbh. 

We have a pretty image focused project (construction site) and I tried everything to fix load times.. I must admit that I got pretty fast response with correct use of Image component layouts and loadings, but still, It would be great if I could control even more.

**Reply by tbanys** [reply]
I admire next but image component only usable in 10.0.7 version, after that images loading in ages and that's very disappointing. Tried to update to newer versions but always have to downgrade to 10.0.7

**Reply by simplenotezy** [reply]
I have the same issue. Considering looking into 10.0.7 or imgix. It's weird it's so slow. 10-15 seconds sometimes.

**Reply by deadcoder0904** [reply]
@Sandstedt it's not a server problem. Image component makes my 1 image take ~30 seconds to load on localhost. It's pretty unintuitive to use.

The API should be a pure replacement for `<img />` tag so we can swap it easily. You can see how people are frustrated with it here → https://github.com/vercel/next.js/issues/23637

**Reply by Sandstedt** [reply]
I know, I haven't said anything otherwise. That's why we have switched to ImgIx handling the image generation instead of the current node sollution. In the meanwhile. This obviously need to be addressed.

**Reply by ghost** [reply]
just add package `yarn add sharp`

For any queries
https://nextjs.org/docs/messages/sharp-missing-in-production

It works amazing

**Reply by dagda1** [reply]
i tried that but nothing happened

**Reply by Alannek1992** [reply]
I have the same problem.. the initial load of the images takes incredibly long even though I have included sharp etc. next:12.0.9. Any ideas? After the initial load the images are probably cached and the problem is gone.

**Reply by stanislavyv** [reply]
> just add package `yarn add sharp`
> 
> For any queries https://nextjs.org/docs/messages/sharp-missing-in-production
> 
> It works amazing

Still unreasonably slow, even with sharp

**Comment by kristianstroka** [comment]
Any updates on this? ... there are more new versions, but still same issue with images ... :(

**Reply by deadcoder0904** [reply]
@kristianstroka nope, subscribe to this issue → https://github.com/vercel/next.js/issues/23637

**Comment by simplenotezy** [comment]
I believe a workaround could be if you could modify the "cache-control" header. In such case I could have my Cloudflare Worker proxy cache the image, since I don't expect them to change ever really.

**Comment by dagda1** [comment]
I have been experiencing this.  I just waited 20 seconds while demoing to a client for the images to appear.  I was very embarrased

**Reply by Sandstedt** [reply]
@dagda1 Been there. Feel the pain.

Just upgrade to next 11 or 12 and make sure you run `npm install sharp` (and make sure it's installed on the server too), and you will be all set (it's so much faster then the default arm-optimized image processing).

And set minimum cache ttl in next.config.js:
```
images: {
    minimumCacheTTL: 31536000,
}
```

**Comment by dagda1** [comment]
@Sandstedt thank you.  I'm on next 11 and I have the sharp package installed in `"dependencies"`.

How would it not be installed on the server?  That sounds like something that might be happening.

I'll update my config to use the `minimumCacheTTL`.

**Reply by Sandstedt** [reply]
> @Sandstedt thank you. I'm on next 11 and I have the sharp package installed in `"dependencies"`.
> 
> How would it not be installed on the server? That sounds like something that might be happening.
> 
> I'll update my config to use the `minimumCacheTTL`.

Run `npm ls sharp` in the project root on the server to see if it's installed.

Else maybe monitor the CPU performance to see if it's a bottleneck there. We're usually fine with the lowest tier on Digital Oceans dedicated servers.

Also make sure that the images gets a long cache time in the http headers. A year is probably good at least. Else the generated images won't be saved.

You can also make sure you don't delete the .next/images folder (maybe it's .next/cache/images or something) while ex rsync'ing up files to the server. Else you will delete all previous generated files on the server when deploying a new version.

**Comment by dagda1** [comment]
@Sandstedt I tried the above suggestions but they did not improve anything that much.

I just removed all `next/image` from my repo and moved all static content delivery to cloudfront and here are the results:

<img width="1381" alt="cdn" src="https://user-images.githubusercontent.com/118328/147549147-411842d1-2079-409f-a40a-c513fdcab19c.png">

`next/image` was a horrendous experience, I wasted a lot of time trying to make this work and for no reason.  I find a lot of vercel opaque and the advice was to use the `sharp` package but I have no idea whether it was working or not and I definitely used it but there was no feedback to say whether it worked or not.

I love how easy vercel is to deploy via git push.  I love the temporary preview URLs although I do find this unreliable but for images and static asset serving, I found it pretty awful and no real way to make things better outside of next.config.

**Reply by Sandstedt** [reply]
Oh, I have not idea if my suggestion works in Vercel. I was talking about e regular deployment to our own servers. (We're using Digital Ocean and Ubuntu servers). But yeah, there is zero feedback on what image engine is in use, more than without Sharp it's super slow on Linux servers.

But great you solved it in another way.

**Comment by dagda1** [comment]
the `sharp` thing is totally opaque. It appears to work for some but not others.  There are no logs or way of knowing what is happening.

I ended up putting my images in AWS cloudfront

**Reply by BelalKhalifaKP** [reply]
@dagda1 How do you load you images from the CDN, I got the same problem, but with remote images. They seem to load twice idk what's wrong and the sharp thing is not really working. The CDN itself is extremely fast, but then the image loads for a second time for seconds.

**Comment by tooshikii** [comment]
I experience the same issue on my app deployed on AWS, amplify although I'm setting up sharp as an image loader with environmental variables set up as cited in the issue here.
https://github.com/vercel/next.js/issues/23637 

Has anyone encounterd the same issue and found a sloution ?

**Reply by MonkeyHenry** [reply]
@tooshikii Did you solve your problem? I'm also using AWS Amplify and tried a bunch of different things suggested in this thread to no avail.

**Comment by Alireza17224** [comment]
Same problem

**Comment by Tosinkoa** [comment]
Pls is there any solution yet? am also facing this problem

**Comment by Alireza17224** [comment]
To fix this problem, first put the most important images to 
`
priority={true}
`


this prop will delete the lazy load from your image and it will be much faster!

Also installing sharp can improve this action

yarn add sharp

npm install sharp

**Reply by atomoc** [reply]
genius

**Comment by MONTYYYYY** [comment]
same. here  next js version: 12.3.1

**Comment by M1ck0** [comment]
I am facing this issue since next 10 I think, no one seemed to care then, and no one seems to care now. Sad

**Comment by lazyflog** [comment]
next image -> img is working to me!

**Comment by techinp** [comment]
I have stopped to use `next/image` change to use `<img/>` instead while waiting for fix this problem!

**Reply by dagda1** [reply]
Do you know vercel actually charge for image optimization.  I hope it is not based on this solution  because it does not work for a lot of people and the whole process is totally opaque

**Comment by M1ck0** [comment]
Looks like the issue is less noticeable on the new version of Next and `next/image`.

**Comment by JohnnyMcFadden** [comment]
Decided to just move back to img, doesn't seem worth the pain of debugging. Images load pretty fast client side from an s3 bucket and cdn for me.

**Comment by cbdeveloper** [comment]
Yeah, I'm also getting very low loading speeds on my Next v12 app (haven't tried adding `sharp` yet).
I just go the news that Vercel will start charging for optimized images. They got to have a solution for this.
If `sharp` doesn't fix it, either I'll go back to `<img/>` or I'll use some other image optimization tool.

**Comment by dagda1** [comment]
@cbdeveloper here is the awful response I got from a vercel customer support person when I linked to this thread:

> The thead you mentioned is almost several years old now and in the Next.js discussion section, not the Vercel section. We have many customers using our image optimization service without issues and in my first email I checked and confirmed it's currently working as expected.

> If it's not working for yourself, then it could be down to a project configuration or code issue on the project 

Basically he tells me it works on his machine and it is my fault.

**Reply by cbdeveloper** [reply]
@dagda1 , that's sad.

Yesterday I got really annoyed by this and I tried a couple of things that definitely helped. At first, I was basically just rendering the images and setting `priority` to the ones above the fold. But even those ones were taking a long to time to load when you'd refresh the page.

My only custom config for image on `next.config.js` was:

```
images: {
    domains: ["images-storage-domain.com"]
  },
```

Then I decided to do this (based on some discussions I found here and on Stack Overflow):

- Add `sharp` as a project dependency
- Add `minimumCacheTTL: 31536000`

Then my config was now:

```
images: {
    domains: ["images-storage-domain.com"],
    minimumCacheTTL: 31536000
  },
```

Not sure if the `sharp` dependency did anything to help or not (I deployed both changes at once), but I guess that the `minimumCacheTTL` did most of the improvement. Because now I only get a slow loading once per image. If I refresh the website now, I get a `x-vercel-cache: HIT` on the images I've previously loaded, and they show up pretty fast.

But keep in mind you'll get a slow loading once, per image, **per media width parameter**. Example:

```
_next/image?url=__IMAGE_URL__%3Falt%3Dmedia&w=640&q=75
_next/image?url=__IMAGE_URL__%3Falt%3Dmedia&w=1080&q=75
_next/image?url=__IMAGE_URL__%3Falt%3Dmedia&w=1920&q=75
```

Then, at least one user on `mobile, tablet and desktop`, will get the slow loading.

---

I'm still having issues with the `cache-control` header, though.

**From:** https://nextjs.org/docs/api-reference/next/legacy/image

![image](https://user-images.githubusercontent.com/43407798/209276846-f3070300-8b96-4791-a2d1-67c6899b1e73.png)

Even though they say that `minimumCacheTTL` should take precedence if it's larger than the cache-control of the original upstream image, I'm still getting this for the optimized images:

![image](https://user-images.githubusercontent.com/43407798/209277056-e8bdb3cd-4b06-4a1c-8b36-d69470e4b784.png)

Not sure how to solve this, as I definitely don't want my media storage having to worry about setting `Cache-Control`. This should be doable from the `NextJS` server, which is the one serving the images to the client.

**Alternative I'm considering:**

https://cloudinary.com/documentation/media_optimizer

**Comment by stefann01** [comment]
Same problem on my side with next v13. I load aroung 15-20 images from a google cloud bucket and display them in a grid [for a photography portfolio]. With `unoptimized prop set to true` images loaded faster for some reason. After seeing the thread discussion here I installed sharp and it `loads much faster without unoptimized set to true`. 
My conclusion is that the built in optimizer is way to slow.

It would be nice to have this specified in the docs... ☹️

**Comment by julian624** [comment]
I have the same issue, im searching a solution since 3 weeks ago and can't find anything

**Reply by mananbordia** [reply]
Did you find any solution ?

**Comment by vtassios** [comment]
Same issue.

I have next 12. I get my images through VM on Digital Ocean. The first time someone visits the page has slow image load. After that, the images cached and they are loading much faster. I am using `next/Image` but `<img />` doesn't have any difference. Is there anything I can do through code or is just a matter of VM ?

Is there any legit solution ?

**Comment by kesar** [comment]
same here.

<img width="527" alt="Captura de pantalla 2023-01-31 a las 12 21 37" src="https://user-images.githubusercontent.com/1288106/215746773-26b7136c-1f24-4efc-8a6b-ca05bff1128e.png">

**Comment by M1ck0** [comment]
This issue has been fixed in the latest versions. Also, make sure that you have `sharp` installed

https://www.npmjs.com/package/sharp

**Reply by eyvindxt** [reply]
Could you please elaborate on "latest versions"? Is for example 12.3.4 from 3 months ago included there?

**Reply by M1ck0** [reply]
I've been using `13.1.6` and it works as it should. I also have `sharp` installed.

This is from docs
> Starting with Next.js 13, the next/image component was rewritten to improve both the performance and developer experience. In order to provide a backwards compatible upgrade solution, the old next/image was renamed to next/legacy/image.

https://nextjs.org/docs/api-reference/next/legacy/image

**Reply by stanislavyv** [reply]
I have 14.2.3 and it's embarrassingly slow on first load. Sometimes it refuses to load at all even after like 20 secs and it's only fixed on page refresh...

**Comment by dagda1** [comment]
I think vercel are in denial on thus one.  I've had sharp installed forever but I  get the same lazy answer that it works on my machine.  The whole thing is opaque with no feedback if I'm getting oprimization or not apart from how God  damn slow the images are to load.  To charge for this is a joke and I'm going to move to remix and be done with next which is very slow on lots of counts

**Reply by M1ck0** [reply]
I've been using `13.1.6` and it works as it should. I also have `sharp` installed.

This is from docs
> Starting with Next.js 13, the next/image component was rewritten to improve both the performance and developer experience. In order to provide a backwards compatible upgrade solution, the old next/image was renamed to next/legacy/image.

https://nextjs.org/docs/api-reference/next/legacy/image

**Reply by Charlio99** [reply]
> I've been using `13.1.6` and it works as it should. I also have `sharp` installed.
> 
> This is from docs
> 
> > Starting with Next.js 13, the next/image component was rewritten to improve both the performance and developer experience. In order to provide a backwards compatible upgrade solution, the old next/image was renamed to next/legacy/image.
> 
> https://nextjs.org/docs/api-reference/next/legacy/image

I have this exact same version and slow as hell for me, only works decent when images are cached by the browser

**Reply by M1ck0** [reply]
@Charlio99 

Are you testing it in production? Is there a link that I can visit to check network requests? Do you have `sharp` installed? Are you using `next/image` or `next/legacy/image`?

**Reply by Charlio99** [reply]
> @Charlio99
> 
> Are you testing it in production? Is there a link that I can visit to check network requests? Do you have `sharp` installed? Are you using `next/image` or `next/legacy/image`?

Yes, it's a production environment, and I'm using next/image + sharp.
I emailed you with the website (I want to avoid sharing it here because it's still under development)

**Reply by Charlio99** [reply]
> > @Charlio99
> > Are you testing it in production? Is there a link that I can visit to check network requests? Do you have `sharp` installed? Are you using `next/image` or `next/legacy/image`?
> 
> Yes, it's a production environment, and I'm using next/image + sharp. I emailed you with the website (I want to avoid sharing it here because it's still under development)

I'm almost 100% sure it's not server related because I have 3 other more frontends on the same server and none of them has issues.

**Reply by kesar** [reply]
> I'm almost 100% sure it's not server related because I have 3 other more frontends on the same server and none of them has issues.

I feel it might be CDN related, because I am based in Spain as well and get crazy response times

**Reply by Charlio99** [reply]
> > I'm almost 100% sure it's not server related because I have 3 other more frontends on the same server and none of them has issues.
> 
> I feel it might be CDN related, because I am based in Spain as well and get crazy response times

Server is in the Netherlands, and I'm using local storage to avoid S3 or other image hosting possible issues. I also tried connecting from a VPN in the Netherlands and I get the same result

**Reply by M1ck0** [reply]
Can you share the `<Image/>` code line with me? One that loads a big cover image on the website. You don't have to share image path.

**Reply by Charlio99** [reply]
> Can you share the `<Image/>` code line with me? One that loads a big cover image on the website. You don't have to share image path.

```
<Image
  src="SOMELOCALPATH.WEBP"
  layout="fill"
  loading="eager"
  priority={true}
  quality={100}
  objectFit="cover"
  objectPosition={"bottom"}
  alt=""
  className="absolute inset-0"
  draggable="false"
/>
```

**Reply by Charlio99** [reply]
But it's not only the big image, the products images are also very slow and even the small SVG region flag in the top right corner

**Reply by Charlio99** [reply]
@M1ck0 can you try to load the website now (incognito, erasing cookies, cache, etc) and let me know the load times?

**Reply by M1ck0** [reply]
Loading the initial image is much faster now.

This code sample that you sent is not from the new `next/image`. This is `next/legacy/image` as half of these params are not available anymore, or have been renamed.

Also, don't use `loading` prop as it can degrade performance. You should use `priority`

https://nextjs.org/docs/api-reference/next/image#loading

**Reply by Charlio99** [reply]
> Loading of the initial image is much faster now.
> 
> This code sample that you sent is not from the new `next/image`. This is `next/legacy/image` as half of these params are not available anymore, or have been renamed.

Is there anything to do besides upgrading to next13? I updated from next12 to next13 last week just bumping the next version in the package.json (and the requirements)

**Reply by M1ck0** [reply]
```jsx
<Image
  src="SOMELOCALPATH.WEBP"
  fill
  priority
  alt=""
  className="absolute inset-0"
  draggable="false"
/>
```

`objectFit` and `objectPosition` are deprecated now. You should set them through `css`. Try this and tell me if it works any better.

Read more [here](https://nextjs.org/docs/api-reference/next/image)

**Reply by Charlio99** [reply]
> ```js
> <Image
>   src="SOMELOCALPATH.WEBP"
>   fill
>   priority
>   alt=""
>   className="absolute inset-0"
>   draggable="false"
> />
> ```
> 
> `objectFit` and `objectPosition` are deprecated now. You should set them through `css`. Try this and tell me if it works any better.
> 
> Read more [here](https://nextjs.org/docs/api-reference/next/image)

Still slow. When the web loaded faster what I did is to rollback to next12

**Reply by sedouard** [reply]
> Still slow. When the web loaded faster what I did is to rollback to next12

Same here. I've upgraded to Next 13 and ensured sharp is installed and available. I even set `NEXT_SHARP_PATH` to the absolute path of the sharp NPM directory.

**Comment by winchell** [comment]
There are almost certainly multiple causes in this thread but this bug disappeared when I stopped creating my own `server.js` that included custom launch code like `new NextServer({`. When I let the build create the `server.js` (as designed -- this should be the norm), the Image component works as expected.

**Comment by jsandovalg** [comment]


**Reply by M1ck0** [reply]
Putting `unoptimized={true}` is the same as if you are default HTML `<img/>` tag

> When true, the source image will be served as-is instead of changing quality, size, or format. Defaults to false.

**Comment by consal** [comment]
I use the recent nextjs 13, deploy on Vercel. My images are around 200kb, I have Embla carousel with the images. Sometimes the images take a couple of seconds to load... Does anyone have a working solution? I installed sharp, it didn't change anything.

**Reply by xie392** [reply]
Me too, have you solved this problem now?

**Comment by jserrao** [comment]
Hacky, but you could pretty easily build your own image component and then try to load all of your images via something like S3. You can basically just go around the whole optimization if you want.

**Reply by quoctruongdev** [reply]
I am also facing the same problem in next js 13.4.12.

**Comment by damirdjukanovic** [comment]
Facing the same issue, I've removed the avif from formats in next.config.js and it does load faster, but still a bit slow. 

`images: {
formats: ['image/webp']
}`

**Reply by stanislavyv** [reply]
They do mention that avif takes 20% more time to generate, but still...

**Comment by bluengreen** [comment]
I've done numerous a/b comparisons and still not understanding where vercel gets this notion that next/image makes images load "faster". Thats just not true. I suppose if I uploaded a photo directly from my camera, full size, it might be faster as it scales and optimizes that image. I've found just doing some pre-processing and/or using a CDN, (cloudfront, cloudflare, etc) is soooooooo much faster its not even remotely close. I'm using Bunny CDN and optimizing by passing a query string to set the size, crop etc, but any of these inline CDN processors will work. It's obscene the difference in speed. 

Another thing to consider is next/image **requires** a serverless function to run. Even if you create a separate loader. I found this out when they sent me a message saying that images will no longer load cause I blew past the 1000 invocation limit in the free vercel hosting. I re-wrote my loader into my own wrapper function and voila - no more limits. 

Also, you want to be mindful on where you use next/image. Because of these constraints like the function invocation, and requiring window before rendering the image, your design layout will shift as the images become available to load. So for something critical like your logo in the upper left corner you will want to just use the plain old `img` tag with an optimized image. The browser will handle it so much faster.

**Comment by McJezuss** [comment]
## Please remember to check Cloudflare DNS settings!

I use Cloudflare as my DNS manager but never disabled the proxy on the DNS records for Vercel. For some reason, this caused images to load insanely slow (I'm talking upwards of 10s on some images that were already pre-optimized with quality set to 60%).

However, when the proxy is disabled on the DNS records for Vercel, the images load nearly instantly, even after a full reload and on new devices.

I hope this helps someone out there.

**Reply by Prakash-Growmax** [reply]
where should i disable in the vercel

**Comment by stanislavyv** [comment]
Still very slow in 14.2.3. I tried both `Image` and `getImageProps`. App in ECS with sharp installed. I'm using `priority` for LCPs and also `sizes` where appropriate. `quality` on most images is set to 75 (default). `width` and `height` I get from the image metadata. Once loaded it works well, but after a fresh deployment some images take up to 20+ seconds to load.

**Reply by ramyNiranjan** [reply]
Hi 
I have the same setup as you. Did you find an answer to it, or how is your image loading time now?

**Reply by ariunaamy** [reply]
Were you able to figure this out?

**Comment by duylddev** [comment]
same issue when images are store in AWS S3 (same region with backend),
image is 75kb but take 3seconds to "wait for backend response"
any solution?

**Reply by bluengreen** [reply]
If you are using the next/image component its calling a serverless function on every request. Skip the image component and just use a html `img` tag, and a cloudfront distribution. Try that.

**Reply by duylddev** [reply]
I'm using aws ec2, not vercel hosting
want to use image component for auto resize feature
but if next/image is not ready for production, I'll go back to the old way

**Reply by bluengreen** [reply]
I wouldn't say its not ready for production but its for optimization. Your image most likely doesn't need the optimization cause its so small. The extra server hit is unnecessary.

**Reply by duylddev** [reply]
oh, I got your point
my bad when using next/image, look likes it will make s3 cloudfront become useless (cause now it always hit our server, not cdn)
thanks a lot :D

**Reply by bluengreen** [reply]
Yeah thats the idea. However, next/image doesn't actually cache. I find s3 > cloudfront is faster. I also spun up `imgproxy` on aws lambda using this template. Gonna check that out so it will sit inbetween s3 > imgproxy > cloudfront. 

https://imgproxy.net/blog/almost-free-image-processing-with-imgproxy-and-aws-lambda/

**Comment by BelalKhalifaKP** [comment]
Absolutely! The magic is in the aspect ratio of the Image. When you have an
Image with a size of 200x300 and you don't set the aspect ratio to 2-3 in
className or set width and height to 200 and 300 it will be slow. Really
just resizing Images changes everything. If you wanna display the image in
the size 150x250 and set the width and height to 150 and 250 but the image
itself has another aspect ratio, you will have problems. Just resize the
image to a ratio that fits 150-250.

Let me know if this changed your life.

Am Mi., 13. Nov. 2024 um 13:36 Uhr schrieb Ariunaa Myagmar <
***@***.***>:

**Reply by ariunaamy** [reply]
Hey, thanks! Adding the exact sizing to the class name helped a
bit, but the issue is  is that the images in the database are managed by
someone else, and they upload various images with different ratios. We
currently have around 2,000 images in the database, so I’m considering
running a script in the background to resize them to a uniform aspect ratio. 
And then I am sure it will get better.  Or if you have better idea lmk

**Comment by BelalKhalifaKP** [comment]
Azure CDN was also a gamechanger for me. Can you check the image loading
time. It’s important to know how long it gets requested from the db and how
long it takes to be rendered afterwards.

**Comment by zr00da** [comment]
```
import MyImage from '@/public/images/image.webp'

<Image
  src={MyImage}
  width={1280}
/>
```

Imagine the surprise when I found out the above makes my static site always download a 1.5MB avif at 2x the source resolution marked as a slow request (runs a serverless function?) instead of the 200kb static webp I provided, with `max-age=0` caching as a bonus...

I don't understand the defaults here - why is a `srcset` generated at all when I didn't provide it? Why does the component think I want a resample that's bigger than the source (that it knows the size of) which only magnifies the asset size without any improvement in quality since it's not an AI upscale? Why aren't these assets generated statically during SSG build? What's up with the magic behavior changes around the `fill` property that really just sets some CSS around `object-fit`? And the cache headers - is it not reasonable to trigger cache invalidation with a url param if a new build sees the asset actually change instead?

I can see the more useful usecases in the docs, but when the barebones functionality can set you up for a serious performance hit in what I'd call an elementary usecase then isn't it reasonable to say the component might be defaulting to too many magic assumptions about the platform and srcset generation specifically?

**Reply by dmythro** [reply]
This is so odd. Is it still reproducing?

**Comment by BitBeaver-182** [comment]
Not sure if everyone keep having this issue, but I had it and struggle a while with it. 
I'm currently running in a standalone output on an EC2 instance (just to test things out).  
My issues where that even when I had sharp installed it prompted me the message: `Sharp Missing In Production` even when I had it installed. and when running it images would load painfully slow. I did the following:
Run in your projects directory:
```bash
export NEXT_SHARP_PATH=$(pwd)/node_modules/sharp
```

Run a build
```bash
npm run build
```

copy the assets as recommended in the [docs](https://nextjs.org/docs/pages/api-reference/config/next-config-js/output#automatically-copying-traced-files).

run your server.js

```bash
node .next/standalone/server.js
```

1st checkoint:
this should already not give you any errors in your console:

moving on..

**moving files to the server.**

here, whatever you're using it's okey. rsync is my prefer choice for this quick tests.

move the output folder into your server. this can be the command you run if you choose rsync:
```bash
rsync -avz --delete ./.next/standalone/   user@host:/path/to/your/folder
```

Once in the server you can either run it as is with the comand above or I prefer to use systemctl 

```bash
[Unit]
Description=Your awesome project
After=network.target

[Service]
User=your-user
WorkingDirectory=/path/to/your/folder
# Ensure that the production environment is set and NEXT_SHARP_PATH is defined.
Environment="NODE_ENV=production"
Environment="NEXT_SHARP_PATH=/path/to/your/folder/node_modules/sharp"
# If you use nvm, load it (or set the PATH so node can be found)
# just type which node and it should give you the path to put bellow
ExecStart=/path/to/node server.js.  
Restart=always
RestartSec=10

[Install]
WantedBy=multi-user.target
```

save it as whatever server name you want .service inside the `/etc/systemd/system` path.
the run

```bash
sudo systemctl daemon-reload
sudo systemctl restart your-service-name.service
sudo systemctl status your-service-name.service
```

I hope this helps at least 1 person.

**Reply by mkanavakatini** [reply]
Just putting this here for visibility, I'm using `pnpm` and had to add this in my `package.json`
```
  "pnpm": {
    "onlyBuiltDependencies": [
      "sharp"
    ]
  }
  ```

**Comment by den-ii** [comment]
It's not about the server, it's  as if nextjs Image renders late. I have found something that works for me though. I got a real low quality of my image, compressed it and converted it to base64 string, then I was able to use it like this:
It works great for me.
```
 <Image
            src={"/signup.png"}
            alt="creator"
            layout="fill"
            objectFit="cover"
            loading="eager"
            priority
            className="rounded-bl-2xl rounded-tl-2xl"
            blurDataURL={signupImgBlur}
            placeholder="blur"
          />
```

If  you're rendering something like an icon it's best you use @svgr/webpack package for svg's
